src/main/java/org/ibp/api/java/impl/middleware/germplasm/GermplasmServiceImpl.java

```diff
@@ -313,15 +327,26 @@ private void populateGermplasmPedigree(final List<GermplasmDTO> germplasmDTOList
 	}
 
 	@Override
-	public List<AttributeDTO> getAttributesByGid(
-		final String gid, final List<String> attributeDbIds, final Integer pageSize, final Integer pageNumber) {
-		this.validateGidAndAttributes(gid, attributeDbIds);
-		return this.germplasmDataManager.getAttributesByGid(gid, attributeDbIds, pageSize, pageNumber);
+	public List<AttributeDTO> getAttributesByGUID(
+			final String germplasmUUID, final List<String> attributeDbIds, final Integer pageSize, final Integer pageNumber) {
+		this.validateGuidAndAttributes(germplasmUUID, attributeDbIds);
```

Comment:
1. Change MW to `gettAttributesByGUID` so you don't need to retrieve by GUID just to get GID. 
2. For posterity, you don't need the `isPresent` checking since you've already validated the GUID exists
3. Remove throwing of ApiRuntimeExeption("invalid germplasmDbId).